src/fabric.c

```diff
@@ -485,10 +498,45 @@ void ofi_create_filter(struct fi_filter *filter, const char *raw_filter)
 		++raw_filter;
 	}
 
-	filter->names= ofi_split_and_alloc(raw_filter, ",", NULL);
+	filter->names = ofi_split_and_alloc(raw_filter, ",", NULL);
 	if (!filter->names)
 		FI_WARN(&core_prov, FI_LOG_CORE,
 			"unable to parse filter from: %s\n", raw_filter);
+
+	tkn_str = strtok((char *)raw_filter, ";");
+	if (!tkn_str)
+		return;
+
+	while (tkn_str != NULL) {
+		count_prov_names++;
+		tkn_prove_names[i++] = tkn_str;
+		tkn_str = strtok (NULL, ";");
+	}
+
+	for(i = 0; i < count_prov_names; i++) {
+		if(!ofi_getprov(tkn_prove_names[i], strlen(tkn_prove_names[i]))) {
+			FI_WARN(&core_prov, FI_LOG_CORE,
+				"provider %s is unknown, misspelled or DL provider?\n",
+				tkn_prove_names[i]);
+			not_found_names++;
```

Comment:
We have the mismatched name here.  Call a function that takes the unmatched name as input and looks for the best match.

Reply:
@shefty  Done.